Remake analog circuitry

filcarv opened the pull request:
Ready for final validation of schematics and layout.

Not ready:
. BOM
. Pnp
. Gerbers
. Cross validation

filcarv committed:
Removed level shifters

. Removed level shifters LS_CLK0 & LS_DAC0 & LS_ADC0.
. Removed it's decoupling caps C75 & C72 & C71.
. Level-shifters IO lines connected & routing made.

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

filcarv committed:
Changed DC/DC output to ±6V

. DC/DC stage dimensioned to ±6V output
. Used parallel/series resistors that already exist as OEPS PN codes

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

filcarv committed:
Updated -5V_REG0 to output -5.3V

. Changed LDOs designators
. Feedback resistors have 0.1% tolerance

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

filcarv committed:
Replaced LDOs and update feedback resistors accordingly.

. Replaced digital and analog 5V LDOs
. Replaced with NJM11100F1-TE2
. Feedback resistors calculated & placed but still without OEPS PN
. New and replaced passives with 0603 cases
. PCB routing not yet consistent with Schematic (to be made)

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

filcarv committed:
Added a new power rail of +3.3VA to supply ADCs and DACs.

. Added new power rail 3V3
. Replaced previous ADC and DAC supplies with the new power rail
. Completed layout of all the current changes
. Added MPN and OEPS PN to all components

Not completed yet:
. Take care of names/silk in the new layout

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

filcarv committed:
Added calculations for the analog input ampops.

filcarv committed:
Reshape entire analog input and output.

. Reshaped analog circuit around ADC ampops.
. Reshaped analog circuit around DAC ampops.
. Renamed old FPGA exposed pin names to new ones.
. BNC Clock-out returned to 5V logic.
. MPN optimized to minimize variety.
. Added test points to all the internal voltage rails.

REMAINING
. Add production files
. Cross check BOM with PnP

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

jonnew reviewed (commented):
The bias and gain network for the DAC amplifiers seems wrong:

![image](https://github.com/user-attachments/assets/0ca99b33-ef49-46dc-80f6-11936b651ab8)

To map Vdac = 0 to -5V and Vdac = 3.3 to 5V, you need Vdac_ref = 2.46V and R1/R2  = 2.03 in the opamp feedback networks. Here is the worked arithmetic and simulation:

![image](https://github.com/user-attachments/assets/aa6846f5-a818-4910-8992-d004e38aa93a)

![image](https://github.com/user-attachments/assets/49a57e29-f724-46c5-a565-325f08e3241e)

![image](https://github.com/user-attachments/assets/22397300-6318-4094-b606-2e69bb106ec6)

jonnew commented:
There are several global labels that are not connected in the schematic. 

![image](https://github.com/user-attachments/assets/c6dd5be2-c5b1-46a4-a7bd-2c2ad680f4af)

If this is intentional, get rid of them and replace with a NC flag:

![image](https://github.com/user-attachments/assets/8e931051-631d-43c3-a038-cb89837534e6)

jonnew commented:
What is the purpose of this resistor on all the ADC input amplifiers?

![image](https://github.com/user-attachments/assets/b94f7bcf-9ff8-4b04-aaaf-a950c68abde2)

the non inverting input can be tied to ground.

jonnew committed:
Pull request review

- Minor schematic cleanup
- Add proper sheets to both schematic and pcb
- Add DAC amplifier spice

jonnew reviewed (changes requested):
Please pull the changes I committed and have a look at my comments.

filcarv commented:
> What is the purpose of this resistor on all the ADC input amplifiers?
> 
> the non inverting input can be tied to ground.

It's a parallel 6.19k // 10k , it helps reduce noise. It's mandatory but since we had the case already there and connected from previous revision, why not?

filcarv committed:
Updated LTspice circuits according to current schematic.

filcarv committed:
Fixed DAC circuit.

filcarv committed:
Fix not connected pins and re-order available power rails.

Co-Authored-By: David Palma <78764277+davidpalma666@users.noreply.github.com>

jonnew commented:
> > What is the purpose of this resistor on all the ADC input amplifiers?
> > the non inverting input can be tied to ground.
> 
> It's a parallel 6.19k // 10k , it helps reduce noise. It's mandatory but since we had the case already there and connected from previous revision, why not?

How does this reduce noise? Maybe im missing something, but resistors are wide band noise sources.  So adding it here is equivalent to adding a little thermal noise source to a high impedance node.

Here is a noise analysis of the circuit from 0.1 ohms (green line) to 10k (dark red). 

![image](https://github.com/user-attachments/assets/547e2b5b-559e-4d4f-b0f1-0d003bcfc5d0)

this is exactly what i would expect -- the wide band noise increases as the resistor value increase. The bandwidth of the opamp is applied to this white noise the same for every value. 


EDIT: I see -- its not for noise, its for compensating for bias current and resulting DC offsets. It is to compensate for the combine voltage drop across the feedback network by creating an equal voltage across this resistor. 

It seems like this practice was more relevant when these bias currents were substantial.  it also requires the input offset current to be near zero. So, its up to you.

EDIT 2: The typical input bias current for this opamp is 100 nA. Its offset bias current is 10 nA. 

This bias results in V = 100e-9 * 3.82k ~ 0.4 mV offset with +/- 0.04 mV due to variations in bias current.

But, the **input offset voltage** of this device is ±10 mV!

So this circuit does not do much except slightly increase the noise. Thanks for the lesson though, this was good to know. Feel free to keep this if you want because it does not do much of anything.

jonnew committed:
Fixed schematic/pcb parity

- Manually annotate schematic

jonnew commented:
```
One thing: I think you have copy&paste schematic from one sheet to other and the sch <> pcb is not synchronized anymore.
Do you want us to do that or do you prefer to do it yourself because you know better what you've done and the risk is lower?
``` 

This was caused by the insane use of reference designators on this design. In the future, we must start with a proper schematic. It should be fixed in the latest commit.

jonnew committed:
More back annotation from PCB to schematic

- There should be no parity errors now
- Lots of warnings about footprints still
- The footprints have not courtyards. This is bad for pick and place
manufacturability.

filcarv committed:
Prepared files for BOM.

. Add description and fix some minor issues.
. Increase IAR of J1 to make the board cheaper.

filcarv committed:
Changed to allow .csv files.

filcarv committed:
Added production files.
